### PR TITLE
MuSig2: Add session cleanup RPC

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/6494 by adding a `MuSig2Cleanup` RPC.

### CODE REVIEW

File: lnwallet/rpcwallet/rpcwallet.go [hunk 1]
```diff
@@ -782,6 +782,28 @@ func (r *RPCKeyRing) MuSig2CombineSig(sessionID input.MuSig2SessionID,
...
+	ctxt, cancel := context.WithTimeout(context.Background(), r.rpcTimeout)
+	defer cancel()
+
+	_, err := r.signerClient.MuSig2Cleanup(ctxt, req)
+	if err != nil {
+		err = fmt.Errorf("error cleaning up MuSig2 session in remote "+
+			"signer instance: %v", err)
+
+		// Log as critical as we should shut down if there is no signer.
+		log.Criticalf("RPC signer error: %v", err)
```
Comment (bhandras) [inline lnwallet/rpcwallet/rpcwallet.go:800]: Is the critical error justified here? We could just call with an invalid id and it'd trigger a restart.

Comment (guggero) [reply lnwallet/rpcwallet/rpcwallet.go:800]: Ah, good catch! I guess that goes for all the forwarded RPC calls. I think we'll need something like a `shutdownIfUnavailable()` call that only logs the `Criticalf` message if it's a gRPC connection error.

Comment (bhandras) [reply lnwallet/rpcwallet/rpcwallet.go:800]: Yeah I think I overlooked those when reviewing. The proposed solution sgtm.

**Comment by alexbosworth** [comment]
Do signing sessions ever expire themselves or only on restart?

**Comment by Roasbeef** [comment]
> Do signing sessions ever expire themselves or only on restart?

Only on restart atm, they have no built in expiration.

File: lntest/itest/lnd_taproot_test.go [hunk 2]
```diff
@@ -906,7 +906,14 @@ func testTaprootMuSig2CombinedLeafKeySpend(ctxt context.Context, t *harnessTest,
...
 			MessageDigest: sigHash,
-			Cleanup:       true,
+		},
+	)
+	require.NoError(t.t, err)
+
+	// We manually clean up session 3, just to make sure that works as well.
+	_, err = alice.SignerClient.MuSig2Cleanup(
+		ctxt, &signrpc.MuSig2CleanupRequest{
+			SessionId: sessResp3.SessionId,
```
Comment (Roasbeef) [inline lntest/itest/lnd_taproot_test.go:916]: Add an assertion that trying to do another signature w/ the same session ID fails?

**Review by Roasbeef — APPROVED** [review]
LGTM 🪘

Only comment to add an additional assertion after the session clean up.

File: lnwallet/rpcwallet/rpcwallet.go (hunk 1 above)
Comment (guggero) [reply lnwallet/rpcwallet/rpcwallet.go]: Added a `considerShutdown` to the error case of all remote signing RPC calls.

**Review by bhandras — APPROVED** [review]
LGTM :1st_place_medal:

**Pull request merged**